kitsune/wiki/facets.py

```diff
@@ -138,13 +121,21 @@ def _db_documents_for(locale, topics=None, products=None):
                 url=d.get_absolute_url(),
                 document_parent_id=d.parent_id,
                 document_summary=d.current_revision.summary,
+                display_order=d.display_order,
+                helpful_votes=votes_dict.get(d.current_revision_id, 0),
             )
         )
 
+    # sort the results by ascending display_order and descending votes
+    # sort is stable in python, so we sort by votes first, then display_order
+    doc_dicts.sort(key=itemgetter("helpful_votes"), reverse=True)
```

Comment:
nit: you can use `sorted` with two keys

Reply:
I don't see this in the docs, do you have to use a lambda? https://docs.python.org/3/library/functions.html#sorted

Reply:
or pass a func to the `key`

Reply:
Something worth noting, by using `sorted` I had anything between 6-8x better results when I was timing it.
I used something like
```python
doc_dicts = sorted(doc_dicts, key=lambda item: (-item('helpful_votes'], item['display_order']))
```

Reply:
I didn't realize you could do `<` comparisons between tuples like that in python, neat.

I don't think the speedup has anything to do with sort/sorted, indeed sort is slightly faster according to the docs, I guess because it's doing the sort in place (which we're ok with here):

> You can also use the list.sort() method. It modifies the list in-place (and returns None to avoid confusion). Usually it’s less convenient than sorted() - but if you don’t need the original list, it’s slightly more efficient.

*from https://docs.python.org/3/howto/sorting.html#sorting-basics*

I also imagine you got a speedup from your lambda sorting the list differently to the two sorts after each other. The tuple should be `(item['display_order'], -item['helpful_votes'])`.

Regardless of any performance improvements, the lambda is a lot more readable than two sorts and a comment explaining that sort is stable, so I'll switch to that.